### PR TITLE
Fix duplicated root state name on absolute transition in V4 XState

### PR DESCRIPTION
**Pull request by danielxbauer**
Fixed: Absolute Transitions `TransitionTo.Absolute("root", "xyz")` gets serialized as `#root.root.xyz` instead of `#root.xyz`

### CODE REVIEW

File: Statecharts.NET.Core/Interfaces/Context.cs [hunk 1]
```diff
@@ -1,8 +1,6 @@
-﻿using System;
-
-namespace Statecharts.NET.Interfaces
+﻿namespace Statecharts.NET.Interfaces
 {
-    public interface IContext<TImplementing> : IEquatable<TImplementing>
+    public interface IContext<TImplementing>
```
Comment (bemayr) [inline Statecharts.NET.Core/Interfaces/Context.cs:3]: please remove this from this PR 🙈
I wanna start the PRs as clean as possible 👌

File: Statecharts.NET.XState/XStateExtensions.cs [hunk 2]
```diff
@@ -156,20 +156,16 @@ JSProperty Guarded(string @event, IEnumerable<Target> targets, Option<JSProperty
...
-        internal static SimpleValue Serialize<TContext>(
+        internal static SimpleValue Serialize(
             this Target target,
-            string sourceStatenodeName,
-            StatechartDefinition<TContext> statechartDefinition)
-            where TContext : IContext<TContext>
+            string sourceStatenodeName)
             => SimpleValue(target.Match(
-                absolute => $"#{statechartDefinition.Id}." + string.Join(".", absolute.Id.Values),
+                absolute => $"#{string.Join(".", absolute.Id.Values)}",
```
Comment (bemayr) [inline Statecharts.NET.XState/XStateExtensions.cs:163]: Thanks so much! 👏

**Review by bemayr — CHANGES_REQUESTED** [review]
Thanks a lot Daniel for finding this, I had it in mind but forgot about it 🙈, I just want to do the `IEquatable`-stuff in its own PR.

**Comment by danielxbauer** [comment]
@bemayr I reverted the "Remove IEquatable" change

**Comment by danielxbauer** [comment]
@bemayr Sorry, I did something wrong. I close this PR and make 2 separate ones.
See: #4 and #5

**Pull request closed without merging**